src/components/Modals/Modal.vue

```diff
@@ -110,6 +113,14 @@ export default {
 
       // clear input and close modal when complete
       dropdown.value = '';
+      this.closeCurrentModal();
+    },
+    addSemester() {
+      const seasonInput = document.getElementById(`season-placeholder`);
+      const yearInput = document.getElementById(`year-placeholder`);
+
+      this.$parent.addSemester(seasonInput.innerHTML.split(' ')[1], parseInt(yearInput.innerHTML, 10));
+
```

Comment:
Why are we reading raw DOM html here? Vue should already have provided good solution to avoid this.

Reply:
Fixing of this is more involved. It depends on better cross-component communication. Hacky HTML is a non-solution.